responsiveness added

Solodanna opened the pull request:
feat(responsive-design): implement mobile-first layout with media queries and grid refactor

- Set desktop-first as default styling approach
- Added comment annotations to organize CSS by device type (desktop/tablet/mobile)
- Introduced media queries for tablet and mobile breakpoints
- Modified color (background-color) of the navigation panel (nav)
- Tablet styles and Mobile styles:
  - Updated header (h1) font size and color
  - Updated image (img) border color and box-shadow color
  - Modified style of the buttons (.nav-button & .nav-button:hover) - color and font style (font-weight, color)
- Refactored 'Experience' section from flexbox to CSS grid
- Implementing flexible and scalable CSS units (rem, fr) for sizing of the design elements

Solodanna committed:
responsiveness added

juddin1995 commented on css/index.css:
```diff
@@ -1,3 +1,5 @@
+/* Desktop Version - Basic Style */
```
Please explicitly state that this is desktop-first Layout. Your comment should read 
// Used desktop-first layout

juddin1995 commented on css/index.css:
```diff
@@ -134,15 +141,103 @@ li {
+  margin-top: 0.625rem;
+}
+
+/* Tablet Version (768px - 1024px) */
```
Great job identifying breakpoints for your media query. You should also add a comment after the closing brace of the media query for both tablet and mobile version.

juddin1995 commented on css/index.css:
```diff
@@ -134,15 +141,103 @@ li {
+    font-size: 0.9rem;
+  }
+}
+
```
please add 
/* End of Tablet Version Styles */

juddin1995 commented on css/index.css:
```diff
@@ -134,15 +141,103 @@ li {
+    flex-direction: column;
+    align-items: center;
+  }
 }
```
Please add the following after this line
please add
/* End of Mobile Version Styles */

juddin1995 reviewed (approved):
### Achievements

1. Well done on setting up the media queries for both tablet and mobile versions with relevant style changes. You correctly made a minimum of two style property changes for multiple HTML elements in each media query.

2. The HTML file is well structured, and your use of semantic tags is commendable.

3. You have also successfully implemented a grid layout for the Experience section, showcasing understanding of CSS Grid, which fulfills one of the stretch goals.

4. Your use of consistent CSS practices, such as flexbox and grid, is noticeable and demonstrates a solid grasp of layout principles.

5. The overall look of the website is clean and consistent, showing your attention to design principles.

### Areas for Improvement

1. Determine if you'd like to build your site mobile-first or desktop-first. While your CSS code is well structured, it would be helpful to explicitly state in a comment at the beginning of your CSS file whether you are designing mobile-first or desktop-first. This will clarify your approach and help keep your code organized.

2. Add comment notation to help keep yourself organized and make it easier to find your desktop/tablet/mobile style code sections. While you did add comments for the beginning of the layouts, it would enhance readability to clearly label each section with comments indicating where the styles for desktop, tablet, and mobile versions end more explicitly. Currently, it's noted only where the desktop/mobile/tablet styles begin.

3. You should add responsiveness to the navigation menu layout so that on smaller screens the nav will become a burger menu. Check out this video for an example: [Create a Responsive Hamburger Menu using only HTML and CSS | Responsive Navigation Menu](https://youtu.be/QQlxvj_GKss
)
4. Be wary of using bright text on bright backgrounds. Please check these contrast guidelines for further information -- [Contrast Guidelines](https://www.w3.org/WAI/WCAG22/Techniques/general/G18). Use this tool to check if your site meets the contrast requirements:
[Contrast Checker](https://webaim.org/resources/contrastchecker/)
This is an easier to read guide: [Accessibility: Color Contrast](https://www.w3schools.com/accessibility/accessibility_color_contrast.php)

Solodanna committed:
- Adjusted styles for better visibility
  (using darker colors for text and lighter for backgrounds).
- Determined the code as desktop-first.
- Added comment notation for desktop/tablet/mobile style code sections.
- Added a hamburger menu button for better navigation on smaller screens.
